crates/bevy_reflect/src/type_path.rs

```diff
@@ -161,3 +167,58 @@ impl<T: TypePath> DynamicTypePath for T {
         Self::module_path()
     }
 }
+
+/// Provides dynamic access to all methods on [`TypePath`].
+#[derive(Clone, Copy)]
+pub struct TypePathVtable {
+    // Cache the type path as it is likely the only one that will be used.
+    type_path: &'static str,
+    short_type_path: fn() -> &'static str,
+    type_ident: fn() -> Option<&'static str>,
+    crate_name: fn() -> Option<&'static str>,
+    module_path: fn() -> Option<&'static str>,
```

Comment:
I wonder if the increased memory footprint would worth it to cache all of these values, so as to avoid any performance cost caused by reading from a `RwLock` (on my machine doing so is a 8–11ns operation).

Probably not, right?

Reply:
these are pretty much never called and are just there for completeness so i don't think it would be worth it.

`short_type_path` is the only one that i think is ever called internally.